pyro/infer/mcmc/nuts.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import absolute_import, division, print_function
+
+from collections import namedtuple
+
+import torch
+
+import pyro
+import pyro.distributions as dist
+from pyro.infer.util import torch_data_sum
+from pyro.ops.integrator import velocity_verlet
+from pyro.util import ng_ones, ng_zeros
+
+from .hmc import HMC
+
+
+_TreeInfo = namedtuple("TreeInfo", ["z_left", "r_left", "z_right", "r_right",
+                                    "z_proposal", "size", "turning", "diverging"])
+
+
+class NUTS(HMC):
+    """
+    No-U-Turn Sampler kernel, where ``step_size`` need to be explicitly specified by the user.
```

Comment:
Can you add a little more information here, and maybe a small instantiation example?  I know the HMC documentation is a bit slim as well but since I suspect this will get a fair amount of use once it's mature it should be made as accessible as possible.

Reply:
Yes, I'll do.